packages/x-scheduler/src/joy/internals/components/time-grid/TimeGrid.tsx

```diff
@@ -0,0 +1,161 @@
+    }
+    const hasScroll = body.scrollHeight > body.clientHeight;
+    header.style.setProperty('--has-scroll', hasScroll ? '1' : '0');
+  }, [events]);
+
+  const lastIsWeekend = isWeekend(adapter, days[days.length - 1]);
+
+  const handleHeaderClick = React.useCallback(
+    (day: SchedulerValidDate) => (event: React.MouseEvent) => {
+      onDayHeaderClick?.(day, event);
+    },
+    [onDayHeaderClick],
+  );
+
+  const renderHeaderContent = (day: SchedulerValidDate) => (
+    <span className="TimeGridHeaderContent">
+      {/* TODO: Add the 3 letter week day format to the adapter */}
+      <span className="TimeGridHeaderDayName">{adapter.formatByString(day, 'ccc')}</span>
+      <span className="TimeGridHeaderDayNumber">{adapter.format(day, 'dayOfMonth')}</span>
+    </span>
+  );
+
+  return (
+    <div ref={forwardedRef} className={clsx('TimeGridContainer', 'joy', className)} {...other}>
+      <TimeGridPrimitive.Root className="TimeGridRoot">
+        <div ref={headerWrapperRef} className="TimeGridHeader">
+          <div className="TimeGridGridRow TimeGridHeaderRow" role="row">
+            <div className="TimeGridAllDayEventsCell" />
+            {days.map((day) => (
+              <div
+                key={day.day.toString()}
+                id={`TimeGridHeaderCell-${day.day.toString()}`}
+                role="columnheader"
+                aria-label={`${adapter.format(day, 'weekday')} ${adapter.format(day, 'dayOfMonth')}`}
+              >
+                {onDayHeaderClick ? (
+                  <button
+                    type="button"
+                    className="TimeGridHeaderButton"
+                    onClick={handleHeaderClick(day)}
+                    tabIndex={0}
+                  >
+                    {renderHeaderContent(day)}
+                  </button>
+                ) : (
+                  renderHeaderContent(day)
+                )}
+              </div>
+            ))}
+          </div>
+          <div
+            className={clsx('TimeGridGridRow', 'TimeGridAllDayEventsRow')}
+            role="row"
+            data-weekend={lastIsWeekend ? '' : undefined}
+          >
+            <div
+              className="TimeGridAllDayEventsCell TimeGridAllDayEventsHeaderCell"
+              role="columnheader"
+            >
+              {translations.allDay}
+            </div>
+            {days.map((day) => (
+              <div
+                key={day.day.toString()}
+                className="TimeGridAllDayEventsCell"
+                aria-labelledby={`TimeGridHeaderCell-${day.day.toString()}`}
+                role="gridcell"
+                data-weekend={isWeekend(adapter, day) ? '' : undefined}
+              />
+            ))}
+          </div>
+        </div>
+        <div ref={bodyRef} className="TimeGridBody">
+          <div className="TimeGridScrollableContent">
+            <div className="TimeGridTimeAxis" aria-hidden="true">
+              {Array.from({ length: 24 }, (_, hour) => (
```

Comment:
Nitpick: This will need custom handling to cover DST days. 🙈 
Could be worth keeping a TODO comment. 🤔